src/tilda/migration/Migrator.java

```diff
@@ -320,43 +329,46 @@ else if (A.getClass() == DDLDependencyPostManagement.class)
+                        ;
+                        
                         // We have to check if someone changed goal-posts for VARCHAR and CLOG thresholds.
                         // The case here is that we have a CHAR(10) in the database, and the model still says
                         // STRING/10, but the thresholds have changed in such a way that now, it should be in the DB
                         // as VARCHAR(10). We have to check that the "final" type from the model is consistent with
                         // the type in the DB. The previous set of checks look at fundamental type changes, for example
                         // from INT to STRING etc... But they won't catch an internal change of CHAR to VARCHAR not due to
                         // model changes, but to threshold changes.
-                        if (Col.isCollection() == false && Col.getType() == ColumnType.STRING
+                        boolean condition2 = Col.isCollection() == false && Col.getType() == ColumnType.STRING
                         && (CMeta._TypeSql.equals("CHAR") == true && C.getDBStringType(Col._Size) != DBStringType.CHARACTER
-                        || CMeta._TypeSql.equals("VARCHAR") == true && C.getDBStringType(CMeta._Size) == DBStringType.CHARACTER))
-                          {
-                            Actions.add(new ColumnAlterType(C, CMeta, Col));
+                        || CMeta._TypeSql.equals("VARCHAR") == true && C.getDBStringType(CMeta._Size) == DBStringType.CHARACTER)
+                        ;
+                        
+                        if (condition1 || condition2)
+                          {                      
+                            CAM.addColumnAlterType(CMeta, Col);
                             NeedsDdlDependencyManagement = true;
                           }
                         // Else, we could still have a size change and stay within a single STRING DB type
-                        else if (Col.isCollection() == false && Col.getType() == ColumnType.STRING)
+                        if (!condition2 && Col.isCollection() == false && Col.getType() == ColumnType.STRING)
                           {
                             DBStringType DBStrType = C.getDBStringType(CMeta._Size);
                             if (DBStrType != DBStringType.TEXT && CMeta._Size != Col._Size)
                               {
-                                Actions.add(new ColumnAlterStringSize(CMeta, Col));
+                                CAM.addColumnAlterStringSize(CMeta, Col);
                                 NeedsDdlDependencyManagement = true;
                               }
                           }
                         if (CMeta._Nullable == 1 && Col._Nullable == false || CMeta._Nullable == 0 && Col._Nullable == true)
                           Actions.add(new ColumnAlterNull(Col));

```

Comment:
Investigate

Reply:
This is just altering the nullable field of the column at the end of the object processing. It was never part of the else statement directly preceding it. I've added back the else statement.